enhancements/sig-architecture/224-event-based-manifestwork/README.md

```diff
@@ -0,0 +1,788 @@
+                "image": "busybox:1.31",
+                "imagePullPolicy": "IfNotPresent",
+                "name": "busybox",
+                "ports": [{"containerPort": 80}]
+            }]}
+        }
+    }
+}],
+"manifestConfigs": [{
+    "resourceIdentifier": {
+        "group": "apps",
+        "resource": "deployments",
+        "name": "busybox-48150",
+        "namespace": "default"
+    },
+    "feedbackRules": [
+        {
+        "type": "JSONPaths",
+        "jsonPaths": [{"name": "status", "path": ".status"}]
+        }
+    ]
+  }]
+}
+```
+
+2. Updating resource - the mwrs-hub-controller can send a cloud event to the topic `/sources/sdbdeavo-mwrs-hub-controller/clusters/cluster1/manifests`:
+
+```JSON5
+// Context Attributes
+{
+  "specversion": "1.0",
+  "type": "io.open-cluster-management.v1.manifestwork.spec.updated",
+  "source": "/sources/sd3ded4-mwrs-hub-controller/clusters/cluster1/manifests",
+  "id": "35dc1966-1447-49f4-95ae-7fba6017a4fd",
+  "time": "2023-07-19T03:02:11.548189454Z",
+  "datacontenttype": "application/json",
+}
+// Extensions
+{
+  "resourceid": "a52adbe8-b6f2-52c8-9378-c4f544502fb7",
+  "resourceversion": 2
+}
+// Data
+{
+"manifests": [{
+    "apiVersion": "apps/v1",
+    "kind": "Deployment",
+    "metadata": {
+        "name": "busybox-48150",
+        "namespace": "default"
+    },
+    "spec": {
+        "replicas": 2,
+        "selector": {"matchLabels": {"app": "busybox"}},
+        "template": {
+          "metadata": {
+            "creationTimestamp": null,
+            "labels": {"app": "busybox"}
+          },
+          "spec": {
+            "containers": [{
+                "command": ["sh", "-c", "sleep 3600"],
+                "image": "busybox:1.31",
+                "imagePullPolicy": "IfNotPresent",
+                "name": "busybox",
+                "ports": [{"containerPort": 80}]
+            }]}
+        }
+    }
+}],
+"manifestConfigs": [{
+    "resourceIdentifier": {
+        "group": "apps",
+        "resource": "deployments",
+        "name": "busybox-48150",
+        "namespace": "default"
+    },
+    "feedbackRules": [
+        {
+        "type": "JSONPaths",
+        "jsonPaths": [{"name": "status", "path": ".status"}]
+        }
+    ]
+  }]
+}
+```
+
+3. Deletion resource - the mwrs-hub-controller can send a cloud event to the topic `/sources/sdbdeavo-mwrs-hub-controller/clusters/cluster1/manifests`:
+
+```JSON5
+// Context Attributes
+{
+  "specversion": "1.0",
+  "type": "io.open-cluster-management.v1.manifestwork.spec.deleted",
+  "source": "/sources/sd3ded4-mwrs-hub-controller/clusters/cluster1/manifests",
+  "id": "35dc1966-1447-49f4-95ae-7fba6017a4fd",
+  "time": "2023-07-19T03:03:11.548189454Z",
+  "datacontenttype": "application/json",
+}
+// Extensions
+{
+  "resourceid": "a52adbe8-b6f2-52c8-9378-c4f544502fb7",
+  "resourceversion": 3,
+  "deletiontimestamp": "2023-07-19T03:05:46.841308779Z"
+}
+```
+
```

Comment:
are you talking about the message order? I suggest to use at least QoS 1 and as mentioned to wait for the PUBACK. refer to https://www.hivemq.com/blog/mqtt-essentials-part-6-mqtt-quality-of-service-levels/

Reply:
This depends on implementation? 
Agent should also have a cache for the received manifestwork, if the deleted manifestwork is not found in the cache, do nothing.

Reply:
this depends on the work agent implement, for current implement, if we delete a nonexistent resource, the work agent will ignore it

Reply:
Agreed on delete. I was not talking about message ordering - but thanks for pointing out the PUBACK @clyang82 . 
I guess the bigger question is - how can we scale this work agent by running multiple copies of it ? Refer: https://www.hivemq.com/article/mqtt-client-load-balancing-with-shared-subscriptions/

Reply:
I think we are good here. Agent expects that it will have **`guaranteed & ordered message delivery`** to be honored by the system. Parallelization is a thig we can worry at a later time.